src/axom/sidre/docs/sphinx/query_data.rst

```diff
@@ -0,0 +1,101 @@
+.. ## Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+.. ## other Axom Project Developers. See the top-level LICENSE file for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+
+.. _dataconcepts-label:
+
+===========================
+Datastore Queries
+===========================
+
+Sidre provides a couple of methods that can be called to query certain
+information about the contents of a datastore object. Each method takes
+a Conduit Node and inserts information related to the query into it. The
+caller can then inspect the information by retrieving values of specifically 
+named fields from the Node. This is often helpful for debugging.
+
+-------------
+Buffer Query
+-------------
+
+A datastore object can be queried for information about the buffer objects it
+contains. For example::
+
+  Datastore* ds = ...;
+
+  conduit::Node n;
+  ds->getBufferInfo(n);
+  n.print();   // print Conduit node contents to std out, if desired
+
+This method call inserts four fields into the Node that have numeric values
+accessible as type ``axom::IndexType``. For example::
+
+  using IndexType = axom::IndexType;
+
+  IndexType num_buffers = n["num_buffers"].value();
+  IndexType num_buffers_referenced = n["num_buffers_referenced"].value();
+  IndexType num_buffers_detached = n["num_buffers_detached"].value();
+  IndexType num_bytes_allocated = n["num_bytes_allocated"].value();
+
+The variables have the following values:
+
+  * ``num_buffers`` : Total number of buffer objects in the datastore.
+  * ``num_buffers_referenced`` : Number of buffers referenced by a view (i.e., attached to a view.
+  * ``num_buffers_detached`` : Number of buffers not referenced by a view. Note: ``num_buffers_detached`` = ``num_buffers`` - ``num_buffers_referenced``.
+  * ``num_bytes_allocated`` : Total number of bytes allocated in all buffers.
+
+--------------------
+Group Subtree Query
+--------------------
+
+A group can be queried for information about the data associated with it, or
+with the entire group subtree rooted at the group. For example::
```

Comment:
```suggestion
associated with the entire subtree rooted at the group. For example::
```
Not quite as terse as what you wrote, but I think it is a little easier to understand.

Reply:
change pushed